### DISCUSSION TITLE
HttpListener in Uno

### DISCUSSION THREAD

**Discussion by danbrannanavontus** [unoplatform/uno#8442] [General]
(repost from Uno discord channel)

Hi - what's the preferred way to listen for HTTP requests on the client in uno web assembly? We need to call out to an api which then redirects to a local address which we host. This was done in HttpListener for windows clients however it fails in web assembly as HttpClient isn't supported. Have also tried embedio's http listener implementation but that fails because it uses system.net.sockets which isn't supported. I attempted to use WebHost however, given that microsoft have changed the Api for hosting a web application in pretty much every version of .net core, it's very difficult to use any examples found online. Thanks!

**Comment by jeromelaban** [comment]
`HttpListener` is not supported on WebAssembly because Javascript in the browser does not support it. 

In general, though, redirection is supported by `HttpClient` because the underlying Javascript `fetch` API supports it. Which .NET exception are you getting when your API redirects ?

**Comment by danbrannanavontus** [comment]
Hi, thanks for your reply. 

If I use HttpListener, I get the exception 'HttpListener is not supported on this platform' at the point the listener is started as you suggest. I hadn't tried HttpClient as I had only ever used it for sending requests, not listening and didn't realise that was possible.

With regard to using WebHost and the .net core APIs, I get this: System.MethodAccessException: 'Attempt by method 'Microsoft.Extensions.Logging.Configuration.LoggerProviderConfigurationFactory.GetConfiguration(System.Type)' to access method 'Microsoft.Extensions.Logging.ProviderAliasUtilities.GetAlias(System.Type)' failed.' Googling suggests a version mismatch between Microsoft.Extensions.Logging and the version of .net standard being used (2.0 in this case). However, I am unable to make them match due to other package dependencies in the project.

Thanks

Dan

**Reply by jeromelaban** [reply]
`HttpClient` does not listen, but it is able to handle redirects properly, even if the request goes to a local network endpoint (but still cannot host an http server).

For the other error, it may be [a linker issue](https://platform.uno/docs/articles/get-started-wizard.html#runtime-error-no-parameterless-constructor-defined-for-xxxx), depending on how your application is configured.
